airflow/www/security_manager.py

```diff
@@ -738,24 +633,9 @@ def create_perm_vm_for_all_dag(self) -> None:
     def check_authorization(
         self,
         perms: Sequence[tuple[str, str]] | None = None,
-        dag_id: str | None = None,
     ) -> bool:
         """Checks that the logged in user has the specified permissions."""
         if not perms:
             return True
 
-        for perm in perms:
-            if perm in (
-                (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG),
-                (permissions.ACTION_CAN_EDIT, permissions.RESOURCE_DAG),
-                (permissions.ACTION_CAN_DELETE, permissions.RESOURCE_DAG),
-            ):
-                can_access_all_dags = self.has_access(*perm)
-                if not can_access_all_dags:
-                    action = perm[0]
-                    if not self.can_access_some_dags(action, dag_id):
-                        return False
-            elif not self.has_access(*perm):
-                return False
-
-        return True
+        return all(self.has_access(*perm) for perm in perms)
```

Comment:
Should we also deprecate / raise exception here? See https://github.com/apache/airflow/pull/34317/files#r1355974367

Reply:
Done